data-node/src/main/java/org/graylog/datanode/management/OpensearchRestClient.java

```diff
@@ -35,7 +48,20 @@ public static RestHighLevelClient build(OpensearchConfiguration configuration) {
             if (configuration.authUsername() != null && configuration.authPassword() != null) {
                 credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(configuration.authUsername(), configuration.authPassword()));
             }
-            builder.setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
+
+            try {
+                final var sslContext = SSLContext.getInstance("TLS");
+                final var tm = new CustomCAX509TrustManager(caService);
```

Comment:
`CustomCAX509TrustManager` is bound as Eager Singleton, however, you create it from scratch in this place...
Shouldn't you inject/pass it into this place instead of creating it???